packages/xrpl/src/models/methods/ammInfo.ts

```diff
@@ -0,0 +1,145 @@
+import { Amount, Currency, IssuedCurrencyAmount } from '../common'
+
+import { BaseRequest, BaseResponse } from './baseMethod'
+
+/**
+ * The `amm_info` method gets information about an Automated Market Maker (AMM) instance.
+ * Returns an {@link AMMInfoResponse}.
+ *
+ * @category Requests
+ */
+export interface AMMInfoRequest extends BaseRequest {
+  command: 'amm_info'
+
+  /**
+   * One of the assets of the AMM pool to look up.
+   */
+  asset: Currency
+
+  /**
+   * The other asset of the AMM pool.
+   */
+  asset2: Currency
+}
+
+/**
+ * Response expected from an {@link AMMInfoRequest}.
+ *
+ * @category Responses
+ */
+export interface AMMInfoResponse extends BaseResponse {
+  result: {
+    amm: {
+      /**
+       * The address of the AMM Account.
+       */
+      account: string
+
+      /**
+       * The total amount of one asset in the AMM's pool.
+       * (Note: This could be asset or asset2 from the request)
+       */
+      amount: Amount
+
+      /**
+       * The total amount of the other asset in the AMM's pool.
+       * (Note: This could be asset or asset2 from the request)
+       */
+      amount2: Amount
+
+      /**
+       * (Omitted for XRP) If true, the amount currency is currently frozen for asset.
+       */
+      asset_frozen: boolean
+
+      /**
+       * (Omitted for XRP) If true, the amount currency is currently frozen for asset2.
+       */
+      asset2_frozen: boolean
```

Comment:
If this is omitted for XRP, shouldn't it be optional (with a `?`)?

Reply:
You could theoretically make this object more complex but more specific by specifying that this is included when the asset is not XRP and not when it is

Reply:
Ah yes, it should be with a `?`. It'd rather keep it simple by making them optional :)